lib/addressable/uri.rb

```diff
@@ -560,10 +557,10 @@ def self.normalize_component(component, character_class=
         leave_re = if leave_encoded.length > 0
           character_class = "#{character_class}%" unless character_class.include?('%')
 
-          "|%(?!#{leave_encoded.chars.flat_map do |char|
-            seq = SEQUENCE_ENCODING_TABLE[char]
-            [seq.upcase, seq.downcase]
-          end.join('|')})"
+          bytes = leave_encoded.bytes
+          leave_encoded_downcase = bytes.map { |b| SEQUENCE_ENCODING_TABLE[b] }.join('|')
+          leave_encoded_upcase = bytes.map { |b| SEQUENCE_UPCASED_PERCENT_ENCODING_TABLE[b] }.join('|')
```

Comment:
`SEQUENCE_UPCASED_PERCENT_ENCODING_TABLE` has extra `%`:
```
SEQUENCE_ENCODING_TABLE[
> format("%02x", 45)
=> "2d"
SEQUENCE_UPCASED_PERCENT_ENCODING_TABLE
> format("%%%02X", 45)
=> "%2D"
```
So it should be without the `%` because that's already present outside (`|%`)

Reply:
Hum, I think you are right, but it's weird because I didn't change that 🤔

Reply:
Ah nevermind, I get it now.

Reply:
Should be good now.